userspace/libsinsp/sinsp_filtercheck_k8s.cpp

```diff
@@ -202,81 +202,100 @@ uint8_t* sinsp_filter_check_k8s::extract(sinsp_evt *evt, OUT uint32_t* len, bool
 	{
 		return NULL;
 	}
+
+	// Here we extract info only if we have the container
+	if(tinfo->m_container_id.empty())
+	{
+		return NULL;
+	}
+
+	const auto container_info = m_inspector->m_container_manager.get_container(tinfo->m_container_id);
+	// No labels means no k8s metadata.
+	if(container_info == nullptr || container_info->m_labels.empty())
+	{
+		return NULL;
+	}
+
 	m_tstr.clear();
-	// there is metadata we can pull from the container directly instead of the k8s apiserver
-	const sinsp_container_info::ptr_t container_info =
-		m_inspector->m_container_manager.get_container(tinfo->m_container_id);
-	if(!tinfo->m_container_id.empty() && container_info && !container_info->m_labels.empty())
+
+	switch(m_field_id)
 	{
-		switch(m_field_id)
+	case TYPE_K8S_POD_NAME:
+		if(container_info->m_labels.count("io.kubernetes.pod.name") > 0)
 		{
-		case TYPE_K8S_POD_NAME:
-			if(container_info->m_labels.count("io.kubernetes.pod.name") > 0)
-			{
-				m_tstr = container_info->m_labels.at("io.kubernetes.pod.name");
-				RETURN_EXTRACT_STRING(m_tstr);
-			}
-			break;
-		case TYPE_K8S_NS_NAME:
-			if(container_info->m_labels.count("io.kubernetes.pod.namespace") > 0)
-			{
-				m_tstr = container_info->m_labels.at("io.kubernetes.pod.namespace");
-				RETURN_EXTRACT_STRING(m_tstr);
-			}
-			break;
-		case TYPE_K8S_POD_ID:
-			if(container_info->m_labels.count("io.kubernetes.pod.uid") > 0)
+			m_tstr = container_info->m_labels.at("io.kubernetes.pod.name");
+			RETURN_EXTRACT_STRING(m_tstr);
+		}
+		break;
+	case TYPE_K8S_NS_NAME:
+		if(container_info->m_labels.count("io.kubernetes.pod.namespace") > 0)
+		{
+			m_tstr = container_info->m_labels.at("io.kubernetes.pod.namespace");
+			RETURN_EXTRACT_STRING(m_tstr);
+		}
+		break;
+	case TYPE_K8S_POD_ID:
+		if(container_info->m_labels.count("io.kubernetes.pod.uid") > 0)
+		{
+			m_tstr = container_info->m_labels.at("io.kubernetes.pod.uid");
```

Comment:
@Andreagit97 not related to your changes and we can defer addressing them, but I noticed a few issues in current parsing. 

For example here the pod id should be the truncated (first 12 char) `"io.kubernetes.sandbox.id")`. That way   it matches how we expose the container id.

Reply:
Also the pod id can be also extracted from the ContaienrStatusResponse info field. Happy to help make this more robust in a follow up, not this PR.

```
 "info": {
    "sandboxID"
```

Sharing this as for example I don't see the label `"io.kubernetes.sandbox.id"` in my settings. 

Also am working (finally) on some better unit tests for the container engine, should have it up later today.

Reply:
> For example here the pod id should be the truncated (first 12 char) "io.kubernetes.sandbox.id"). That way it matches how we expose the container id.

I like the idea of having the `pod.id` exposed as the `container.id` but i have 2 questions:
* Are we sure our users don't need the entire `pod.id`? Maybe they use it as it is to correlate some info coming from other sources (external to Falco)
* Are we sure `12` bytes are enough to guarantee the uniqueness of the pod? 

BTW yes, i see this as a follow up of this PR